### PR TITLE
fix type

### PR DESCRIPTION
**Pull request by kazukageorge**

### CODE REVIEW

**Comment by derekk-nm** [comment]
It would be useful to have some tests here that verify this change is working correctly.

File: src/sparsezoo/api/graphql.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def fetch(
         self,
         operation_body: str,
         arguments: Optional[Dict[str, str]] = None,
-        fields: Optional[List[str]] = None,
+        fields: Optional[Union[List[str], Dict[str, Union[Dict, None]]]] = None,
```
[the remark this answers is not in the document]
Comment (kazukageorge) [reply src/sparsezoo/api/graphql.py:31]: graphql will not handle either, but we have a parser that given Dict or List, will convert the input fields to a graphql compatible format.

**Comment by kazukageorge** [comment]
> It would be useful to have some tests here that verify this change is working correctly.

There already are tests to check both List and Dict. 
https://github.com/neuralmagic/sparsezoo/blob/main/tests/sparsezoo/api/test_query_parser.py#L122-L126

**Review by tdg5 — APPROVED** [review]

File: src/sparsezoo/api/graphql.py [hunk 2]
```diff
@@ -72,6 +72,9 @@ def make_request(
             operation_body=operation_body, arguments=arguments, fields=fields
         )
 
+        print(query.arguments)
```
Comment (derekk-nm) [inline src/sparsezoo/api/graphql.py:75]: These lines look like you were debugging?

**Review by derekk-nm — DISMISSED** [review]

File: src/sparsezoo/api/graphql.py [hunk 3]
```diff
@@ -27,8 +27,8 @@ class GraphQLAPI:
     def fetch(
         self,
         operation_body: str,
-        arguments: Optional[Dict[str, str]] = None,
-        fields: Optional[List[str]] = None,
+        arguments: Optional[Dict[str, Any]] = None,
+        fields: Optional[Union[List[str], Dict[str, Union[Dict, None]]]] = None,
```
Comment (tdg5) [inline src/sparsezoo/api/graphql.py:31]: Can you get away with `Optional[Dict]` and avoid the need for one of the `Union`?
```suggestion
        fields: Optional[Union[List[str], Dict[str, Optional[Dict]]]] = None,
```

**Review by tdg5 — DISMISSED** [review]

**Review by derekk-nm — APPROVED** [review]

**Pull request merged**